### PR TITLE
Test suite

### LINKED ISSUE #100: [BUG] # characters inside string literals start a comment
**Environment (please complete the following information):**
 - Rezasm dc62fcb
 - Fedora Linux 39

**Describe the bug**
Any # inside a string literal start a string literal, even though they shouldn't

**Explain Steps To Reproduce**
1. Run `move $s0 "This has a # character`

**Expected behavior**
The # should be a character in the string

### PR DESCRIPTION
**Pull request by AleksBekker**
Some of the Rust code (especially surrounding the test suite) had issues, so I moved around the tests and fixed these issues.

Along the way, I fixed #100 (# character in a string/character literal started a comment), and a few of the tests that were failing. Now, running `cargo test --workspace` executes these tests (successfully, as of the time this PR was opened).

### CODE REVIEW

File: rezasm-source/rezasm-core/src/instructions/implementation/mod.rs [hunk 1]
```diff
@@ -17,3 +17,138 @@ pub fn register_instructions() {
...
+                .get_registers()
+                .get_register("T0")
+                .expect("Register access error")
+                .get_data()
+                .int_value(),
+            121
+        );
+    }
+
+    // Moved from Trevor's test in tests/core.rs
```
Comment (TrevorBrunette) [inline rezasm-source/rezasm-core/src/instructions/implementation/mod.rs:72]: Remove

File: rezasm-source/rezasm-core/src/instructions/implementation/mod.rs [hunk 2]
```diff
@@ -17,3 +17,138 @@ pub fn register_instructions() {
...
+            .deref()
+            .as_any()
+            .downcast_ref::<TestWriter>()
+            .unwrap()
+            .get_data();
+        assert_eq!(output.as_str(), "3\n1.5\nPrint Instructions Work!\n");
+    }
+
+    // FIXME: need to move this test elsewhere; it covers the parser I/O, not simulator I/O
+    // Moved from Trevor's test in tests/core.rs
```
Comment (TrevorBrunette) [inline rezasm-source/rezasm-core/src/instructions/implementation/mod.rs:104]: Create issue for this please and remove comment.

File: rezasm-source/rezasm-core/src/instructions/implementation/mod.rs [hunk 3]
```diff
@@ -17,3 +17,138 @@ pub fn register_instructions() {
...
+        let reg_read: Vec<String> = String::from_utf8(reg_bytes.clone())
+            .unwrap()
+            .lines()
+            .map(|line| line.to_string())
+            .collect();
+
+        // Compare
+        assert_eq!(rezasmfile.lines().expect("failed to get lines"), reg_read);
+
+        // FIXME: need to change seek integers due to added comments in the file
```
Comment (TrevorBrunette) [inline rezasm-source/rezasm-core/src/instructions/implementation/mod.rs:124]: Create issue for this please and remove comment

**Review by TrevorBrunette — CHANGES_REQUESTED** [review]

File: rezasm-source/rezasm-core/src/simulation/simulator.rs [hunk 4]
```diff
@@ -272,3 +279,73 @@ impl Simulator {
...
+                .get_registers()
+                .get_register(&registry::T0.to_string())
+                .unwrap()
+                .get_data()
+                .int_value(),
+            1i64
+        );
+    }
+
+    // Moved from Trevor's test in tests/core.rs
```
[the remark this answers is not in the document]
Comment (AleksBekker) [reply rezasm-source/rezasm-core/src/simulation/simulator.rs:315]: Will do

[the remark this answers is not in the document]
Comment (AleksBekker) [reply rezasm-source/rezasm-core/src/simulation/simulator.rs:315]: Done

File: rezasm-source/rezasm-core/src/simulation/memory.rs [hunk 5]
```diff
@@ -190,3 +190,23 @@ impl Memory {
         Ok(out)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    // Moved from Trevor's test in tests/core.rs
```
[the remark this answers is not in the document]
Comment (AleksBekker) [reply rezasm-source/rezasm-core/src/simulation/memory.rs:198]: Done

**Comment by AleksBekker** [comment]
Now closes #105 @TrevorBrunette @WyattRoss

**Review by TrevorBrunette — APPROVED** [review]

**Pull request merged**